server/manager/src/test/java/org/apache/accumulo/manager/tableOps/compact/CompactionDriverTest.java

```diff
@@ -0,0 +1,135 @@
+    EasyMock.expect(manager.getInstanceID()).andReturn(instance).anyTimes();
+    EasyMock.expect(manager.getContext()).andReturn(ctx);
+    EasyMock.expect(ctx.getZooReaderWriter()).andReturn(zrw);
+
+    final String zCancelID = Constants.ZROOT + "/" + instance + Constants.ZTABLES + "/"
+        + tableId.toString() + Constants.ZTABLE_COMPACT_CANCEL_ID;
+    EasyMock.expect(zrw.getData(zCancelID)).andReturn(Long.toString(cancelId).getBytes());
+
+    PowerMock.replayAll();
+
+    final CompactionDriver driver =
+        new CompactionDriver(compactId, namespaceId, tableId, startRow, endRow);
+    try {
+      driver.isReady(Long.parseLong(tableId.toString()), manager);
+    } catch (AcceptableThriftTableOperationException e) {
+      if (e.getTableId().equals(tableId.toString()) && e.getOp().equals(TableOperation.COMPACT)
+          && e.getType().equals(TableOperationExceptionType.OTHER)
+          && e.getDescription().equals("Compaction canceled")) {
+        // success
+      } else {
+        fail("Unexpected error thrown: " + e.getMessage());
+      }
+    } catch (Exception e) {
+      fail("Unhandled error thrown: " + e.getMessage());
+    }
+    PowerMock.verifyAll();
+  }
+
+  @Test
+  public void testTableBeingDeleted() throws Exception {
+
+    final String instance = "test-instance";
+    final long compactId = 123;
+    final long cancelId = 122;
+    final NamespaceId namespaceId = NamespaceId.of("14");
+    final TableId tableId = TableId.of("43");
+    final byte[] startRow = new byte[0];
+    final byte[] endRow = new byte[0];
+
+    PowerMock.resetAll();
+    Manager manager = PowerMock.createNiceMock(Manager.class);
+    ServerContext ctx = PowerMock.createNiceMock(ServerContext.class);
+    ZooReaderWriter zrw = PowerMock.createNiceMock(ZooReaderWriter.class);
+    EasyMock.expect(manager.getInstanceID()).andReturn(instance).anyTimes();
+    EasyMock.expect(manager.getContext()).andReturn(ctx);
+    EasyMock.expect(ctx.getZooReaderWriter()).andReturn(zrw);
+
+    final String zCancelID = Constants.ZROOT + "/" + instance + Constants.ZTABLES + "/"
+        + tableId.toString() + Constants.ZTABLE_COMPACT_CANCEL_ID;
+    EasyMock.expect(zrw.getData(zCancelID)).andReturn(Long.toString(cancelId).getBytes());
+
+    String deleteMarkerPath = Constants.ZROOT + "/" + instance + Constants.ZTABLES + "/" + tableId
+        + Constants.ZTABLE_DELETE_MARKER;
+    EasyMock.expect(zrw.exists(deleteMarkerPath)).andReturn(true);
+
+    PowerMock.replayAll();
+
+    final CompactionDriver driver =
+        new CompactionDriver(compactId, namespaceId, tableId, startRow, endRow);
+    try {
+      driver.isReady(Long.parseLong(tableId.toString()), manager);
+    } catch (AcceptableThriftTableOperationException e) {
+      if (e.getTableId().equals(tableId.toString()) && e.getOp().equals(TableOperation.COMPACT)
+          && e.getType().equals(TableOperationExceptionType.OTHER)
```

Comment:
Unrelated to the scope of this ticket, but should compaction exceptions get their own `TableOperationExceptionType` in the future?